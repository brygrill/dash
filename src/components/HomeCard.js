// @flow
import React, { Component } from 'react';
import { Card, CardMedia, CardTitle, CardText } from 'material-ui/Card';

export default class HomeCard extends Component {
  props: {
    title: string,
    img: string,
    alt: string,
    imgClass: string,
    content: string,
  };
  render() {
    const { title, img, alt, imgClass, content } = this.props;
    return (
      <Card>
        <CardMedia>
          <img src={img} alt={alt} className={imgClass} />
        </CardMedia>
        <CardTitle title={title} />
        <CardText>
          {content}
        </CardText>
      </Card>
    );
  }
}
