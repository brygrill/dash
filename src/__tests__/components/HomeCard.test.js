import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import HomeCard from '../../components/home/HomeCard';

describe('<HomeCard />', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MuiThemeProvider><HomeCard /></MuiThemeProvider>, div);
  });
  it('Matches snapshot with child', () => {
    const component = renderer.create(
      <MuiThemeProvider><HomeCard /></MuiThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
