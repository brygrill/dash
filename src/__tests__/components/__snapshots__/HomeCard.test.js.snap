// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeCard /> Matches snapshot with child 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 8,
      }
    }
  >
    <div
      className="sbr-home-card-media"
      style={
        Object {
          "backgroundColor": undefined,
          "muiPrepared": true,
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <div
          style={
            Object {
              "maxWidth": "100%",
              "minWidth": "100%",
              "muiPrepared": true,
              "verticalAlign": "top",
              "width": "100%",
            }
          }
        >
          <span
            className={undefined}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#fff",
                "display": "inline-block",
                "fontSize": "6rem",
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
              }
            }
          />
        </div>
      </div>
      
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "block",
            "fontSize": 24,
            "lineHeight": "36px",
            "muiPrepared": true,
          }
        }
      />
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "display": "block",
            "fontSize": 14,
            "muiPrepared": true,
          }
        }
      />
    </div>
    <div
      className="sbr-home-card-content"
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }
    />
  </div>
</div>
`;
